core/src/banking_stage/qos_service.rs

```diff
@@ -747,46 +746,64 @@ mod tests {
             .collect();
         let execute_units_adjustment = 10u64;
 
-        // assert all tx_costs should be applied to cost_tracker if all execution_results are all committed
-        {
-            let qos_service = QosService::new(1);
-            let txs_costs = qos_service.compute_transaction_costs(
-                &FeatureSet::all_enabled(),
-                txs.iter(),
-                std::iter::repeat(Ok(())),
-            );
-            let total_txs_cost: u64 = txs_costs
-                .iter()
-                .map(|cost| cost.as_ref().unwrap().sum())
-                .sum();
-            let (qos_cost_results, _num_included) =
-                qos_service.select_transactions_per_cost(txs.iter(), txs_costs.into_iter(), &bank);
-            assert_eq!(
-                total_txs_cost,
-                bank.read_cost_tracker().unwrap().block_cost()
-            );
-            // all transactions are committed with actual units more than estimated
-            let commited_status: Vec<CommitTransactionDetails> = qos_cost_results
-                .iter()
-                .map(|tx_cost| CommitTransactionDetails::Committed {
-                    compute_units: tx_cost.as_ref().unwrap().bpf_execution_cost
-                        + execute_units_adjustment,
-                })
-                .collect();
-            let final_txs_cost = total_txs_cost + execute_units_adjustment * transaction_count;
-            QosService::update_or_remove_transaction_costs(
-                qos_cost_results.iter(),
-                Some(&commited_status),
-                &bank,
-            );
-            assert_eq!(
-                final_txs_cost,
-                bank.read_cost_tracker().unwrap().block_cost()
-            );
-            assert_eq!(
-                transaction_count,
-                bank.read_cost_tracker().unwrap().transaction_count()
-            );
+        for native_programs_consume_cu_feature_activated in [true, false] {
+            let mut bank = Bank::new_for_tests(&genesis_config);
+            if native_programs_consume_cu_feature_activated {
+                bank.activate_feature(&native_programs_consume_cu::id());
+            } else {
+                bank.deactivate_feature(&native_programs_consume_cu::id());
+            }
+            let bank = Arc::new(bank);
+            // assert all tx_costs should be applied to cost_tracker if all execution_results are all committed
+            {
+                let qos_service = QosService::new(1);
+                let txs_costs = qos_service.compute_transaction_costs(
+                    &FeatureSet::all_enabled(),
```

Comment:
is this test intentionally using inconsistent feature sets?

Reply:
kinda. This is to test `update_or_remove_transaction_costs()`, tx_costs calculated here is just a input, so long it is stable, its actual value doesn't really matter to the test itself (it tests how the "adjustment" is made)